src/settings.c

```diff
@@ -422,6 +459,7 @@ static int read_config(char *config_name, settings_t *settings)
         }
     }
 
+    json_decref(settings_json);
```

Comment:
This may cause problems if `set_xxx_settings` functions take direct references, like strings. Did you check with valgrind if free'd memory isn't accessed?

Reply:
Nevermind, just realised that previous changes address excatly that.